source/driver-models/Compass.cpp

```diff
@@ -75,7 +74,6 @@ void Compass::init(uint16_t id)
 
     // Set a default rate of 50Hz.
     this->samplePeriod = 20;
-    this->configure();
```

Comment:
Removed this call as it wasn't doing anything before.

Basically for all the Compass drivers (FXOS8700, LSM303Mag, MAG3100) they call the `Compass` constructor, which calls this `Compass:init()` function.
Before this PR, as this was called in the Compass constructor (before the derived classes constructor) it would run the base class empty function that does nothing.
I've also checked that all derived classes (FXOS8700, LSM303Mag, MAG3100) do call their `configure()` overwritten method on their constructor.

After this making `configure()` a pure virtual function, having this call in the base class constructor (which technically does nothing) would try to call a method that has not been overwritten yet.